sql/core/src/test/java/test/org/apache/spark/sql/JavaDataFrameSuite.java

```diff
@@ -258,7 +258,9 @@ public void testSampleBy() {
     DataFrame df = context.range(0, 100, 1, 2).select(col("id").mod(3).as("key"));
     DataFrame sampled = df.stat().<Integer>sampleBy("key", ImmutableMap.of(0, 0.1, 1, 0.2), 0L);
     Row[] actual = sampled.groupBy("key").count().orderBy("key").collect();
-    Row[] expected = {RowFactory.create(0, 5), RowFactory.create(1, 8)};
-    Assert.assertArrayEquals(expected, actual);
+    Assert.assertEquals(0, actual[0].getLong(0));
+    Assert.assertTrue(1 <= actual[0].getLong(1) && actual[0].getLong(1) <= 6);
+    Assert.assertEquals(1, actual[1].getLong(0));
+    Assert.assertTrue(4 <= actual[1].getLong(1) && actual[1].getLong(1) <= 11);
```

Comment:
Right now the test returns 6 and 11 right?  I'm just estimating good bounds using a binomial cdf, but 0-8 and 2-13 are reasonable ranges for both given the setup (roughly, less than 0.1% chance something outside that range is OK). Does the same need to be applied in `DataFrameStatSuite` below ideally?